nomad/structs/structs.go

```diff
@@ -3817,6 +3818,129 @@ type TaskEvent struct {
 
 	// GenericSource is the source of a message.
 	GenericSource string
+
+	// DisplayMessage is a human friendly message about the event
+	DisplayMessage string
+
+	// Details is a map with annotated info about the event
+	Details map[string]string
+}
+
+func (event *TaskEvent) PopulateEventDisplayMessage() {
+	// Build up the description based on the event type.
+	if event == nil { //TODO PA needs investigation alloc_runner's Run method sends a nil event when sigterming nomad. Why?
```

Comment:
Hm, interesting. I don't see anywhere in here that `taskDestroyEvent` could still be `nil` by the time `r.destroyTaskRunners(taskDestroyEvent)` is called.

However I do feel like event being nil can intentionally happen (perhaps to cause a sync without changing the task's state?) -- I just can't find where now...

Reply:
I saw it only when shutting down, and it always consistently happened during shutdown. Haven't found the root cause yet..

Reply:
Hm if it's dev mode we destroy/GC all allocations on shutdown, so Nomad would be breaking out of that `OUTER` loop in `AllocRunner.Run` and hitting the `AllocRunner.handleDestroy()` code path if that helps.